server/db/dbcontrollers/tradesController.js

```diff
@@ -56,6 +59,20 @@ module.exports = function (knex) {
           available_cash: available_cash
         });
       })
+      .then(function(resp){
+
+        trade = resp;
+        return module.getPortfolio(userID, matchID);
+
+      })
+      .then(function(port){
```

Comment:
I think the problem that's causing tests to fail might be related to the promise handling here, but I'm not sure yet. I think the then should be off the getPortfolio call rather than off the previous then (since that call itself is returning a promise).

I haven't been able to confirm yet, but I'm working on it. Same would be true on the sell, and it would result in an undefined being returned, so it fits the pattern of issue we're looking for.